providers/class-two-factor-totp.php

```diff
@@ -303,10 +332,12 @@ public function validate_authentication( $user ) {
 	 *
 	 * @param string $key      The share secret key to use.
 	 * @param string $authcode The code to test.
+	 * @param string $hash      The hash used to calculate the code.
+	 * @param int    $time_step The size of the time step.
 	 *
 	 * @return bool Whether the code is valid within the time frame
 	 */
-	public static function is_valid_authcode( $key, $authcode ) {
+	public static function is_valid_authcode( $key, $authcode, $hash = 'sha1', $time_step = 30 ) {
```

Comment:
Should `$hash` default to `self::DEFAULT_CRYPTO` to match what `calc_totp` does? Or maybe it could be a class property with a dedicated getter that passes the value through a filter to enable overrides?

Same with the `$time_step` -- should that default to `self::DEFAULT_TIME_STEP_SEC`?